### PR TITLE
spanner-jdbc: Step 28 - JdbcDatabaseMetadata

### PR DESCRIPTION
**Pull request by olavloite**
Adds JDBC database metadata.

### CODE REVIEW

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java [hunk 1]
```diff
@@ -0,0 +1,1390 @@
...
+  }
+
+  @Override
+  public String getURL() throws SQLException {
+    return connection.getConnectionUrl();
+  }
+
+  @Override
+  public String getUserName() throws SQLException {
+    // TODO: extract from credentials
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java]: still have a todo here?

Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java]: Added implementation for `getUserName()` for known `Credentials` types.

Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java]: I've left the `getFunctions(String, String, String)` method as is (i.e. returning an empty `ResultSet`), as there is no server side equivalent for this in Cloud Spanner, and this is hardly ever used

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java [hunk 2]
```diff
@@ -0,0 +1,1400 @@
...
+    return false;
+  }
+
+  @Override
+  public RowIdLifetime getRowIdLifetime() throws SQLException {
+    return RowIdLifetime.ROWID_UNSUPPORTED;
+  }
+
+  @Override
+  public ResultSet getSchemas(String catalog, String schemaPattern) throws SQLException {
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java]: do we have tests for this (and the other sql query methods)?

Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java]: Yes, but they are created as integration tests (`ITJdbcDatabaseMetaDataTest` in the private repo), so they will be added in a later PR.

**Pull request merged**